scheduler/reconcile.go

```diff
@@ -841,6 +892,119 @@ func (a *allocReconciler) computeStop(group *structs.TaskGroup, nameIndex *alloc
 		}
 	}
 
+	// Handle allocs that might be able to reconnect.
```

Comment:
If we're going thru all the trouble to pull out new methods from big stretches of code, I feel like this should be as well?